File: echo/echo.go
package main

import (
	"fmt"
	"os"
)

func main() {
	argv := os.Args[1:]
	argc := len(argv)
	for i, arg := range argv {
		fmt.Print(arg)
		if i < argc {
			fmt.Print(" ")
		}
	}
	fmt.Println()
}

File: echo/echo.c
#include <stdio.h>

int main(int argc, char *argv[])
{
	argc--;
	int i;
	for (i = 1; i <= argc; i++) {
		fputs(argv[i], stdout);
		if (i < argc) {
			putchar(' ');
		}
	}
	putchar('\n');
	return 0;
}
